api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -178,23 +178,15 @@ message PrivateKeyProvider {
   oneof config_type {
     google.protobuf.Any typed_config = 3 [(udpa.annotations.sensitive) = true];
   }
-}
 
-// [#not-implemented-hide:]
-// Provides a list of private key providers. Envoy will find out an available private
-// key provider from the list on order. If there is none of available private key provider,
-// it may fallback to BoringSSL default implementation based on the `fallback` fallback.
-message PrivateKeyProviderList {
-  // A list of private key providers, and at least one private key provider provided.
-  repeated PrivateKeyProvider private_key_provider = 1 [(validate.rules).repeated = {min_items: 1}];
-
-  // If there is no available private key provider from the list, Envoy will fallback to
-  // the BoringSSL default implementation when the `fallback` is true. The default value
-  // is `false`.
-  bool fallback = 2;
+  // [#not-implemented-hide:]
+  // If the private key provider isn't available (eg. the required hardware capability doesn't existed),
+  // Envoy will fallback to the BoringSSL default implementation when the `fallback` is true.
+  // The default value is `false`.
+  bool fallback = 4;
```

Comment:
nit and just for discussion: would `is_optional` be better? I think there are lots of extension points have similar behavior and `is_optional` is widely used to declare a extension is optional and if the extension is not supported, some default behavior will be enabled.

Reply:
In other place(like in the optional http filter), when we use the `is_optional`, it means it is fine if the extension isn't existed.

But at here, I suppose we only fallback when the provider can't be initialized correctly (like the required hardware capabilities not existed on the host). If the provider isn't existed, I still expect we will reject the config.

Reply:
IMO, it should be OK to use this new field to tell envoy to fallback to default behavior when the provider cannot work (either because hardware capabilities or provider isn't existed).

But your response sounds also reasonable to me. I have no strong point to this. cc @envoyproxy/api-shepherds

Reply:
This seems fine to me.